discovery-provider/src/tasks/index.py

```diff
@@ -243,6 +248,101 @@ def index_blocks(self, db, blocks_list):
     if num_blocks > 0:
         logger.warning(f"index.py | index_blocks | Indexed {num_blocks} blocks")
 
+
+def get_user_track_play_counts(session, user_id, track_id):
+    query = text(
+        f"""
+        select play_item_id, count
+        from aggregate_plays
+        where play_item_id in :ids
+        """
+    )
+    query = query.bindparams(bindparam('ids', expanding=True))
+
+    track_play_counts = session.execute(query, { "ids": track_ids }).fetchall()
+    track_play_dict = dict(track_play_counts)
+    return track_play_dict
+
+# Retrieve the play counts from the identity service
+# NOTE: indexing the plays will eventually be a part of `index_blocks`
+def get_track_plays(self, db):
+    with db.scoped_session() as session:
+        # Get the most retrieved play date in the db to use as an offet for fetching
+        # more play counts from identity
+        most_recent_play_date = session.query(
+                Play.updated_at
+            ).order_by(
+                desc(Play.updated_at),
+                desc(Play.id)
+            ).first()
+        if most_recent_play_date == None:
+            # Make the date way back in the past to get the first play count onwards
+            most_recent_play_date = datetime.datetime(2000,1,1,0,0).timestamp()
+        else:
+            most_recent_play_date = most_recent_play_date[0].timestamp()
+
+        # Create and query identity service endpoint for track play counts
+        identity_url = update_task.shared_config['discprov']['identity_service_url']
+        params = { 'startTime': most_recent_play_date, 'limit': 1000 }
```

Comment:
@jowlee does this only do one request to identity per job? so it'll only catch up by 1000 every time the job runs?

Reply:
Yeah, that's correct. I'm not confident that this will be enough or what the most correct solution should be. 
Do you think it should loop until it returns no more listens (per job)? - The thing w/ that would be if there are continuous listens happening, it might never end.
I think this should be fine though.

Reply:
I'm curious what the number of records this would return without a limit are and how many are added an hour. From that we could calculate catch up time.

Reply:
Also it would be interesting if we only loop if the table is empty until we reach the beginning of the current day and then we revert to the default limit of once per run after that.

Reply:
There are two cases for returning a listen - the anonymous which would be 1 results per track per hour (if the track was listened to in that hour) and the user track listen which is 1 result per user per track if it's be updated after the query date.

Reply:
Currently these are the # of rows per table on prod:
UserTrackListens: 192,692
TrackListenCounts: 812,995

So on init, it would have to cycle through ~1k iterations if it was querying 1k at a time to catch up. 
The limit on the identity `/listens/bulk` endpoint is 5k rn. 
But yeah I think we can probably cycle on init. Thoughts @raymondjacobson

Reply:
So previously when I calculated this, I didn't take into account the TrackListenCounts table. With Just UserTrackListens at 200, I figured this would take 17 min (200,000 / 1,000 * 5s)

Since we have 1m rows, this takes closer to an hour and a half, which isn't totally awful. Maybe we can just try 5K at a time? I was worried about increasing the response to be much larger than that on the identity side, but 5K would make this finish in about 20 min again, right?

Reply:
Perhaps we can just add a 1-liner new query in the health_checks file (plays_check) that returns the last indexed play count timestamp

Reply:
okay this still isn't too bad. if we we 5k at a time we'll catch up quick. plus we can avoid looping logic which would be much. more complex